cql3/expr/expression.cc

```diff
@@ -741,6 +741,34 @@ expression make_conjunction(expression a, expression b) {
     return conjunction{std::move(children)};
 }
 
+static
+void
+do_factorize(std::vector<expression>& factors, expression e) {
+    // A TRUE factor can be eliminated
+    if (auto c = expr::as_if<constant>(&e)) {
+        if (c->type == boolean_type
+                && c->value.is_value()
+                && c->value.view().deserialize<bool>(*c->type) == true) {
```

Comment:
There are also `empty` values. They are neither `null` nor `unset`, but they have 0 bytes of data. Deserialization could potentially break there.

Reply:
We shouldn't let an empty leak into a CQL constant, it can only be inserted into a table.

Reply:
(and we should have more systematic treatment of empties, like converting them to NULLs)